packages/cli/src/commands/mock.ts

```diff
@@ -1,71 +1,41 @@
 import { getHttpOperationsFromResource } from '@stoplight/prism-http';
-import * as signale from 'signale';
+import { pick } from 'lodash';
 import { CommandModule } from 'yargs';
-import { createMultiProcessPrism, CreatePrismOptions, createSingleProcessPrism } from '../util/createServer';
+import { CreateMockServerOptions, createMultiProcessPrism, createSingleProcessPrism } from '../util/createServer';
+import sharedOptions from './sharedOptions';
 
 const mockCommand: CommandModule = {
-  describe: 'Start a mock server with the given spec file',
-  command: 'mock <spec>',
+  describe: 'Start a mock server with the given document file',
+  command: 'mock <document>',
   builder: yargs =>
     yargs
-      .positional('spec', {
-        description: 'Path to a spec file. Can be both a file or a fetchable resource on the web.',
+      .positional('document', {
+        description: 'Path to a document file. Can be both a file or a fetchable resource on the web.',
         type: 'string',
       })
-      .middleware(async argv => argv.operations = await getHttpOperationsFromResource(argv.spec!))
-      .fail((msg, err) => {
-        if (msg) yargs.showHelp();
-        else signale.fatal(err.message);
-
-        process.exit(1);
-      })
+      .middleware(async argv => (argv.operations = await getHttpOperationsFromResource(argv.document!)))
```

Comment:
That looks like a dirty hack that can even fail in certain circumstances.

How about removing that middleware and resolve operations in handler below?

Reply:
I did not find any information that middleware accepts a promise.

Reply:
Looks OK to me, well, in the handler you wouldn't need the mutation I guess. But, why do you think it would fail? The only situation for this to fail would be not having `document` positional argument. But since it is positional it will be checked by `yargs`

Reply:
@StefanDywersant, behold https://github.com/yargs/yargs/pull/1257

Reply:
@lag-of-death that explains everything, thanks!

Reply:
In any case I want to get rid of this middleware and leverage chokidar at some point to see the first read as a "special" reload so we do not have to do this anymore.